source/extensions/quic_listeners/quiche/platform/quic_logging_impl.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include <cerrno>
+#include <cstring>
+#include <iostream>
+#include <sstream>
+
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+
+#include "absl/base/optimization.h"
+
+#define QUIC_LOG_IMPL_INTERNAL(condition, logstream)                                               \
+  switch (static_cast<int>(condition))                                                             \
+  case 1:                                                                                          \
+    logstream
+
+#define QUIC_LOGGER() Envoy::Logger::Registry::getLog(Envoy::Logger::Id::misc)
+
+#define QUIC_LOG_IF_IMPL(severity, condition)                                                      \
+  QUIC_LOG_IMPL_INTERNAL((condition) && quic::IsLogLevelEnabled(quic::severity),                   \
+                         quic::QuicLogEmitter(quic::severity).stream())
+
+#define QUIC_LOG_IMPL(severity) QUIC_LOG_IF_IMPL(severity, true)
+
+#define QUIC_VLOG_IF_IMPL(verbosity, condition)                                                    \
+  QUIC_LOG_IMPL_INTERNAL((condition) && quic::IsVerboseLogEnabled(verbosity),                      \
+                         quic::QuicLogEmitter(quic::INFO).stream())
+
+#define QUIC_VLOG_IMPL(verbosity) QUIC_VLOG_IF_IMPL(verbosity, true)
+
+// TODO(wub): Implement QUIC_LOG_FIRST_N_IMPL.
+#define QUIC_LOG_FIRST_N_IMPL(severity, n) QUIC_LOG_IMPL(severity)
+
+// TODO(wub): Implement QUIC_LOG_EVERY_N_SEC_IMPL.
+#define QUIC_LOG_EVERY_N_SEC_IMPL(severity, seconds) QUIC_LOG_IMPL(severity)
+
+#define QUIC_PLOG_IMPL(severity)                                                                   \
+  QUIC_LOG_IMPL_INTERNAL(quic::IsLogLevelEnabled(quic::severity),                                  \
+                         quic::QuicLogEmitter(quic::severity).SetPerror().stream())
+
+#define QUIC_LOG_INFO_IS_ON_IMPL quic::IsLogLevelEnabled(quic::INFO)
+#define QUIC_LOG_WARNING_IS_ON_IMPL quic::IsLogLevelEnabled(quic::WARNING)
+#define QUIC_LOG_ERROR_IS_ON_IMPL quic::IsLogLevelEnabled(quic::ERROR)
+
```

Comment:
I think we may need TODOs for CHECK and DCHECK macros (including the various (D)CHECK_<condition> variants), since google3 quic_logging_impl.h implicitly provides those as well (by directly #including base/logging.h), and QUICHE code definitely uses them.

Reply:
SG, added a TODO.

Reply:
Thanks